### PR TITLE
Replace MBEDTLS_SSL_HAVE_CBC with PSA_WANT_ALG_CBC_NO_PADDING

### LINKED ISSUE #9335: Replace legacy macros with its `PSA_WANT` counterparts
Replace these macros:
- `MBEDTLS_SSL_HAVE_CBC` to `PSA_WANT_ALG_CBC_NO_PADDING`
- `MBEDTLS_SSL_HAVE_CCM` to `PSA_WANT_ALG_CCM`
- `MBEDTLS_SSL_HAVE_GCM` to `PSA_WANT_ALG_GCM`
- `MBEDTLS_SSL_HAVE_CHACHAPOLY` to `PSA_WANT_ALG_CHACHA20_POLY1305`

The macros should be replaced in all files except `mbedtls_config.h`, `cneck_config.h` and `config_adjust_*.h`.

Ensure that the test are run in the same way.

### PR DESCRIPTION
**Pull request by eleuzi01**
## Description

Partially resolves #9335 

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required, internal changes
- [x] **3.6 backport** not required, needed for 4.0
- [x] **2.28 backport** not required, needed for 4.0
- [x] **tests** not required

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

Help make review efficient:
* Multiple simple commits
  - please structure your PR into a series of small commits, each of which does one thing
* Avoid force-push
  - please do not force-push to update your PR - just add new commit(s)
* See our [Guidelines for Contributors](https://mbed-tls.readthedocs.io/en/latest/reviews/review-for-contributors/) for more details about the review process.

### CODE REVIEW

File: library/ssl_misc.h [hunk 1]
```diff
@@ -254,7 +254,7 @@ uint32_t mbedtls_ssl_get_extension_mask(unsigned int extension_type);
 #if defined(MBEDTLS_SSL_PROTO_TLS1_2)
 
 /* This macro determines whether CBC is supported. */
-#if defined(MBEDTLS_SSL_HAVE_CBC)      &&                                  \
+#if defined(PSA_WANT_ALG_CBC_NO_PADDING)      &&                                  \
```
Comment (gilles-peskine-arm) [inline library/ssl_misc.h:257]: I don't understand why we're doing this [and I don't think we should](https://github.com/Mbed-TLS/mbedtls/issues/9335#issuecomment-2218071604). It's a good thing that SSL have-algorithm guards are separate from crypto supply-algorithm guards.

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
Please put this on hold until we have architectural agreement.

**Comment by gabor-mezei-arm** [comment]
The `MBEDTLS_SSL_HAVE_CBC` macro should not has to be removed.

**Comment by mpg** [comment]
The design discussion has been resolved, see https://github.com/Mbed-TLS/mbedtls/issues/9335#issuecomment-2251107614

**Review by gabor-mezei-arm — APPROVED** [review]
LGTM

**Review by sezrab — APPROVED** [review]
LGTM

**Pull request merged**